pilot/platform/cloudfoundry/servicediscovery.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloudfoundry
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+
+	copilotapi "code.cloudfoundry.org/copilot/api"
+
+	"istio.io/istio/pilot/model"
+)
+
+// CopilotClient defines a local interface for interacting with Cloud Foundry Copilot
+type CopilotClient interface {
+	copilotapi.IstioCopilotClient
+}
+
+// AppPort is the container-side port on which Cloud Foundry Diego applications listen
+const AppPort = 8080
```

Comment:
is 8080 hardcoded for all CF apps?  Or should this be configurable?

Reply:
Yes, for HTTP traffic this is currently hardcoded for all apps in Cloud Foundry.  A future enhancement here might pull this from copilot, but for now its a hardcoded magic number.